src/app/content/utils/seoUtils.ts

```diff
@@ -9,23 +13,125 @@ import {
+    : getParentPrefix(node.parent, intl, includeTitle);
+
+};
+
+const hideMath = (node: Element) => {
+  node.querySelectorAll('math, .os-math-in-para').forEach((el: Element) => {
+    el.outerHTML = '...';
+  });
+  return node;
+};
 
+export const getTextContent = (str: string) => {
+  const parsed = domParser.parseFromString(str, 'text/html');
+  const text = parsed.body.textContent;
+  return text;
 };
 
-export const createTitle = (page: Page, book: Book): string => {
+const removeExcludedContent = (node: Element) => {
+  if (!node) {
+    return null;
+  }
+
+  const excludedSelectors = [
+    // Introductory content
+    '[data-type="abstract"]',
+    '.learning-objectives',
+    '.chapter-objectives',
+    '.be-prepared',
+    '.os-teacher',
+    // End notes and references
+    'section.suggested-reading',
+    'section.references',
+    // Footnote superscript
+    '[role="doc-noteref"]',
+  ];
+
+  const excludedContent = node.querySelectorAll(excludedSelectors.join(',')) || [];
+
+  for (const item of Array.from(excludedContent)) {
+    item.remove();
+  }
+};
+
+export const generateExcerpt = (str: string) => {
+  return str.replace(/\n+/g, ' ').replace(/\s+/g, ' ').trim().substring(0, 152) + '...';
+};
+
+const getParagraphs = (page: HTMLElement) => {
+  const sectionParaSelectors = [
+    '[data-type="page"]>section>p',
+    '[data-type="composite-page"]>section>p',
+    '.intro-body>.intro-text>section>p',
+  ];
+
+  const paraSelectors = [
+    '[data-type="page"]>p',
+    '[data-type="composite-page"]>p',
+    '.intro-body>.intro-text>p',
+  ];
+
+  const sectionParagraphs = Array.from(page.querySelectorAll(sectionParaSelectors.join(',')));
+  const pageParagraphs = Array.from(page.querySelectorAll(paraSelectors.join(',')));
+
+  return [...sectionParagraphs, ...pageParagraphs];
+};
+
+const getPageDescriptionFromContent = (page: HTMLElement): string | null => {
+  if (!page) {
+    return null;
+  }
+  removeExcludedContent(page);
+
+  const paragraphs = getParagraphs(page);
+  const isLongEnough = (node: Element) => node.textContent && node.textContent.length >= 90;
+
+  for (const paragraph of paragraphs) {
+    hideMath(paragraph);
+    if (!isLongEnough(paragraph) || !paragraph.textContent) {
+      continue;
+    }
+    return generateExcerpt(paragraph.textContent);
+  }
+
+  return null;
+};
+
+export const getPageDescription = (services: Pick<AppServices, 'archiveLoader' | 'intl'>, book: Book, page: Page) => {
+  const {intl, archiveLoader} = services;
+  const cleanContent = getCleanContent(book, page, archiveLoader);
+  const doc = domParser.parseFromString(cleanContent, 'text/html');
+  const treeNode = findArchiveTreeNodeById(book.tree, page.id);
+  if (!treeNode) {
+    return '';
+  }
+  const pageNode = doc.body.firstElementChild;
+  const isAppendix = pageNode.classList.contains('appendix');
+  const contentDescription: string | null = isAppendix ? null : getPageDescriptionFromContent(pageNode);
```

Comment:
all the other filtering on content type is happening in `getParagraphs`, i feel like this should be there too. maybe change one of your selectors to add a `:not('.appendix')` on the page node?